### DISCUSSION TITLE
Unable to get running in VS Code

### DISCUSSION THREAD

**Discussion by KShivum** [unoplatform/uno#8512] [Q&A]
Hi, so I haven't used Uno before and I'm trying to get it to run on VS Code (running on Windows without WSL) and I can't seem to get anything to work correctly it seems. So I feel like a bit of information is missing from the 'Get Started on VS Code' For instance it doesn't tell you to install the Uno nuget package (which I found the right one I believe) and I got the project created with the new command. Immediately after following the guide it seems like the VS Code extension doesn't do anything. There is no code completion in the xaml file (even though the button says 'Uno Platform - XAML is running'). In addition, I can't even build and compile the program because In both App.xaml.cs and MainPage.xaml.cs App and MainPage (respectively) does not contain a definition for 'InitializeComponent'. In addition, for the MainPage.xaml.cs (following the guide) The name 'txt' does not exist in the current context.
What am I doing wrong, I have the prerequisites and the only thing that the uno-check fails on is creating an android emulator. Thank you

**Comment by jeromelaban** [comment]
Thanks for reaching out. If you're not getting the intellisense running, you may not have selected the project you're trying to work on in omnisharp.

At the bottom of Code, you'll see either a flame icon, or the name of the active project. If the name ends with `.sln`, this means that you'll need to select a project instead. 

You're also mentioning that you need to install the Uno nuget package, can you explain how you did get around this? Which command did you execute ?

**Reply by KShivum** [reply]
So I have the project set to MyApp.Skia.Gtk, so it isn't that. I can't remember that exact command, but I think it was 'dotnet install UnoPlatformTemplates' or something along those lines. Is there some other way I'm supposed to install it?

**Reply by jeromelaban** [reply]
The templates are supposed to be installed automatically when the Uno Platform VS Code extension is installed. This would indicate another error. Can you show the contents of the `Uno Platform - *` outputs ?

**Reply by KShivum** [reply]
After the errors show up this is what Uno Platform - XAML contains:
```
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: s:\Projects\Uno Test\MyApp
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: s:\Projects\Uno Test\MyApp
info: Uno.LSP.Roslyn.RoslynServerTarget[0]
      Compilation succeeded (_knownNamespaces:2, _typeNamespaceLookup:550)
info: Uno.LSP.Roslyn.RoslynServerTarget[0]
      Waiting for file changes...
```

**Reply by jeromelaban** [reply]
Which errors are you talking about? The `InitializeComponent` related ones?

**Reply by KShivum** [reply]
Yeah the InitializeComponent and the txt doesn't exist (probably the underlying issue)

**Comment by spouliot** [comment]
@KShivum I had the same issue when my vscode/omnisharp was set to use mono (because it was required by a unity project). 

~~After switching my omnisharp configuration to "mono == never" and "Use modern dotnet" == "true" things started to work as I expected.~~

Scratch that :-( It caused a delay but going back to the source file eventually produced the same invalid  errors.

```
[{
	"resource": "/Users/poupou/git/uno/vsc/MyApp/MyApp.Shared/MainPage.xaml.cs",
	"owner": "csharp",
	"code": "CS1061",
	"severity": 8,
	"message": "'MainPage' does not contain a definition for 'InitializeComponent' and no accessible extension method 'InitializeComponent' accepting a first argument of type 'MainPage' could be found (are you missing a using directive or an assembly reference?) [MyApp.Skia.Gtk]",
	"source": "csharp",
	"startLineNumber": 27,
	"startColumn": 9,
	"endLineNumber": 27,
	"endColumn": 28
},{
	"resource": "/Users/poupou/git/uno/vsc/MyApp/MyApp.Shared/MainPage.xaml.cs",
	"owner": "csharp",
	"code": "CS0103",
	"severity": 8,
	"message": "The name 'txt' does not exist in the current context [MyApp.Skia.Gtk]",
	"source": "csharp",
	"startLineNumber": 33,
	"startColumn": 4,
	"endLineNumber": 33,
	"endColumn": 7
}]
```

Still I got both XAML and C# hot reload to work by changing my config.
